system-x/common/src/main/java/software/tnb/common/deployment/OpenshiftDeployable.java

```diff
@@ -18,10 +21,18 @@ default long waitTime() {
     default void deploy() {
         final int retries = 60;
         if (!isDeployed()) {
-            OpenshiftClient.get().createNamespace(); // ensure namespace exists
             create();
         }
         WaitUtils.waitFor(this::isReady, retries, waitTime() / retries,
             "Waiting until the " + this.getClass().getSimpleName() + " resource is ready");
     }
+
+    @Override
+    default void afterAll(ExtensionContext extensionContext) throws Exception {
+        Deployable.super.afterAll(extensionContext);
+        if (TestConfiguration.parallel()) {
+            // In parallel execution, each test class has its own namespace
+            OpenshiftClient.get().deleteNamespace();
```

Comment:
:thinking:  this will probably work fine, but what would happen if 1 test used more than 1 `OpenshiftDeployable` system-x?

Reply:
nice